src/azure-cli-core/azure/cli/core/auth/adal_authentication.py

```diff
@@ -18,9 +19,25 @@ class MSIAuthenticationWrapper(MSIAuthentication):
     def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
         logger.debug("MSIAuthenticationWrapper.get_token invoked by Track 2 SDK with scopes=%s", scopes)
 
+        # In Cloud Shell, we use
+        # - msrestazure to get access token
+        # - MSAL to get VM SSH certificate
         if 'data' in kwargs:
-            from azure.cli.core.azclierror import AuthenticationError
-            raise AuthenticationError("VM SSH currently doesn't support managed identity or Cloud Shell.")
+            if in_cloud_console():
+                import msal
+                from .util import check_result, build_sdk_access_token
+                app = msal.PublicClientApplication(
+                    "placeholder",  # It needs a string placeholder
+                    #token_cache=...,  # TODO: This PoC does not currently maintain a token cache
+                    )
+                result = app.acquire_token_silent(list(scopes),
+                                                  app.get_accounts(username=msal.CURRENT_USER)[0],
+                                                  data=kwargs['data'])
```

Comment:
@jiasli, I'll probably have further experiment on this part, by using a different API, but with same functionality. Meanwhile, you can focus on reviewing the other open questions in this PR. Thanks.

Reply:
@jiasli ^^^